pkg/secrets/fetch_secret.go

```diff
@@ -16,6 +16,8 @@ import (
 	"strings"
 	"time"
 
+	log "github.com/cihub/seelog"
```

Comment:
shouldn't you be using the logging package? Maybe you haven't rebased to the latest master? This should apply to all your logic. `seelog` shouldn't be called directly anymore.